docs/architecture/maui/configuration-management.md

```diff
@@ -1,9 +1,133 @@
+    {
+        get => Preferences.Get(AccessToken, AccessTokenDefault);
+        set => Preferences.Set(AccessToken, value);
+    }
+
+    public bool UseMocks
+    {
+        get => Preferences.Get(IdUseMocks, UseMocksDefault);
+        set => Preferences.Set(IdUseMocks, value);
+    }
+}
+```
+
+Each setting consists of a private key, a private default value, and a public property. The key is always a const string that defines a unique name, with the default value for the setting being a static read-only or constant value of the required type. Providing a default value ensures that a valid value is available if an unset setting is retrieved. This service implementation can be provided via dependency injection to our application for use in view-models or other services throughout the application.
+
+## Data binding to user settings
+
+In the eShopOnContainers mobile app, the `SettingsView` exposes multiple settings the user can configure at runtime. These settings include allowing configuration of whether the app should retrieve data from microservices deployed as Docker containers or whether the app should retrieve data from mock services that don't require an internet connection. When retrieving data from containerized microservices, a base endpoint URL for the microservices must be specified. The image below shows the SettingsView when the user has chosen to retrieve data from containerized microservices.
+
+:::image type="content" source="media/endpoint_settings.png" alt-text="User settings exposed by the eShopOnContainers mobile app":::
+
+Data binding can be used to retrieve and set settings exposed by the `ISettingService` interface. This is achieved by controls on the view binding to view model properties that in turn access properties in the `ISettingService` interface and raising a property changed notification if the value has changed.
+
+The following code example shows the `Entry` control from the `SettingsView` that allows the user to enter a base identity endpoint URL for the containerized microservices:
+
+```xml
+<Entry Text="{Binding IdentityEndpoint, Mode=TwoWay}" />
+```
+
+This `Entry` control binds to the `IdentityEndpoint` property of the `SettingsViewModel` class, using a two-way binding. The following code example shows the `IdentityEndpoint` property:
+
+```csharp
+private readonly ISettingsService _settingsService;
+
+private string _identityEndpoint;
+
+public SettingsViewModel(
+    ILocationService locationService, IAppEnvironmentService appEnvironmentService,
+    IDialogService dialogService, INavigationService navigationService, ISettingsService settingsService)
+    : base(dialogService, navigationService, settingsService)
+{
+    _settingsService = settingsService;
+
+    _identityEndpoint = _settingsService.IdentityEndpointBase;
+}
+
+public string IdentityEndpoint
+{
+    get => _identityEndpoint;
+    set
+    {
+        SetProperty(ref _identityEndpoint, value);
+
+        if (!string.IsNullOrEmpty(value))
```

Comment:
This code tells me, the reader, that whitespace is a valid value. If it is not, please consider using `string.IsNullOrWhitespace` instead.